torch/csrc/jit/script/compiler.cpp

```diff
@@ -569,15 +568,40 @@ struct to_ir {
   }
 
   void runCleanupPasses(std::shared_ptr<Graph>& to_clean) {
+    // NB: moving returns to end needs to run before all other passes
+    moveAllReturnsToEnd(to_clean);
     // remove any uses of tuples that we inserted that are not needed
     LowerSimpleTuples(to_clean);
     ConstantPooling(to_clean);
   }
 
-  FunctionSchema emitDef(
-      const Def& def,
-      const Self& self,
-      Block* block) {
+  // If the graph might not return, at an implicit None return at the end
+  void handleMaybeNoReturn(const Def& def, Block* block) {
+    // if we're emitting a block that isn't the graph block,
+    // (like in a closure), do not add a return
```

Comment:
What about a closure that doesn't end with a return and needs an implicit return?